clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -163,25 +164,43 @@ def DynamicCastOp : CIR_Op<"dyn_cast"> {
       - Otherwise, the operation will return a null pointer value as its result.
 
     The `info` argument gives detailed information about the requested dynamic
-    cast operation.
+    cast operation. It is an optional `#cir.dyn_cast_info` attribute that is
+    only present when the operation models a down-cast or a side-cast.
+
+    The `relative_layout` argument specifies whether the Itanium C++ ABI vtable
+    uses relative layout. It is only meaningful when the operation models a
+    cast-to-complete operation.
   }];
 
   let arguments = (ins DynamicCastKind:$kind,
                        StructPtr:$src,
-                       DynamicCastInfoAttr:$info);
-  let results = (outs StructPtr:$result);
+                       OptionalAttr<DynamicCastInfoAttr>:$info,
+                       UnitAttr:$relative_layout);
+  let results = (outs CIR_PointerType:$result);
 
   let assemblyFormat = [{
-    `(` $kind `,` $src `:` type($src) `,` qualified($info) `)`
-    `->` type($result) attr-dict
+    `(`
+      $kind `,` $src `:` type($src)
+      (`,` qualified($info)^)?
+      (`,` `relative_layout` $relative_layout^)?
+    `)`
+    `->` qualified(type($result)) attr-dict
```

Comment:
We should start adding new operations without parenthesis, but no need to fix it for this PR, just a thing we should start doing in general.